common/src/main/java/com/wynntils/wynn/model/ShamanTotemModel.java

```diff
@@ -0,0 +1,303 @@
+/*
+ * Copyright © Wynntils 2023.
+ * This file is released under AGPLv3. See LICENSE for full license details.
+ */
+package com.wynntils.wynn.model;
+
+import com.wynntils.core.WynntilsMod;
+import com.wynntils.core.components.Managers;
+import com.wynntils.core.components.Model;
+import com.wynntils.mc.event.AddEntityEvent;
+import com.wynntils.mc.event.ChangeCarriedItemEvent;
+import com.wynntils.mc.event.RemoveEntitiesEvent;
+import com.wynntils.mc.event.SetEntityDataEvent;
+import com.wynntils.mc.objects.Location;
+import com.wynntils.mc.utils.McUtils;
+import com.wynntils.wynn.event.CharacterUpdateEvent;
+import com.wynntils.wynn.event.SpellCastEvent;
+import com.wynntils.wynn.event.TotemEvent;
+import com.wynntils.wynn.objects.ShamanTotem;
+import com.wynntils.wynn.objects.SpellType;
+import com.wynntils.wynn.utils.WynnUtils;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import net.minecraft.network.chat.MutableComponent;
+import net.minecraft.network.syncher.SynchedEntityData;
+import net.minecraft.world.entity.Entity;
+import net.minecraft.world.entity.decoration.ArmorStand;
+import net.minecraft.world.item.ItemStack;
+import net.minecraft.world.item.Items;
+import net.minecraft.world.phys.AABB;
+import net.minecraftforge.eventbus.api.SubscribeEvent;
+
+public class ShamanTotemModel extends Model {
+
+    private ShamanTotem totem1 = null;
+    private Integer pendingTotem1Id = null;
+
+    private ShamanTotem totem2 = null;
+    private Integer pendingTotem2Id = null;
+
+    private ShamanTotem totem3 = null;
+    private Integer pendingTotem3Id = null;
```

Comment:
How hard is it to fold these into a list? It would also reduce code duplication.

Reply:
@DonkeyBlaster Are you doing this?

Reply:
I can give it a shot in a day or two, don't have time at the moment

Or we could also merge and the folding into a list could be a separate PR

Reply:
I don't want to merge this as there is a performance concern to me. Ill fix it this afternoon

Reply:
But that is something different to the issue above, right? Code duplication is more a code hygiene concern. From my PoV, this is acceptable. So maybe if bolyai fixes the perf issue, and then we can merge?

And then follow up by de-duping the code above, and making quick spell using the new spell manager.